From 0548b161752150dc369007a3e4ebf2de291976d8 Mon Sep 17 00:00:00 2001
From: goumin <min.gou@spacemit.com>
Date: Fri, 22 Nov 2024 17:53:37 +0800
Subject: [PATCH 1301/1448] Revert "Resolve a warning.Linux 6.6 does not allow
 attempting to synchronously load a module within an asynchronous context."

This reverts commit 5f0419f6462e3ae3a84bffc1a08d93af83bf13f9.

Change-Id: Ifed99d7d9d85def0ff785612fe89a64003afd0a3
---
 drivers/net/ethernet/spacemit/k1x-emac.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/spacemit/k1x-emac.c b/drivers/net/ethernet/spacemit/k1x-emac.c
index 8932171243b4..935b7267a0a5 100644
--- a/drivers/net/ethernet/spacemit/k1x-emac.c
+++ b/drivers/net/ethernet/spacemit/k1x-emac.c
@@ -2875,6 +2875,9 @@ static struct platform_driver emac_driver = {
 		.name = DRIVER_NAME,
 		.of_match_table = of_match_ptr(emac_of_match),
 		.pm     = &k1x_emac_pm_qos,
+#ifdef CONFIG_SPACEMIT_PARALLEL_BOOTING
+		.probe_type = PROBE_PREFER_ASYNCHRONOUS,
+#endif
 	},
 };
 
-- 
2.47.0

